LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -30,19 +29,32 @@ internal sealed class ConcentratorDeduplication<T> :
 
         public ConcentratorDeduplication(
             IMemoryCache cache,
+            IDeduplicationStrategyFactory deduplicationStrategy,
             WebSocketWriterRegistry<StationEui, string> socketRegistry,
-            ILogger<IConcentratorDeduplication<T>> logger)
+            ILogger<IConcentratorDeduplication> logger)
         {
             this.cache = cache;
+            this.deduplicationStrategy = deduplicationStrategy;
             this.socketRegistry = socketRegistry;
             this.logger = logger;
         }
 
-        public bool ShouldDrop(T frame, StationEui stationEui)
+        /// <summary>
+        /// Checks if this class should process the specified request and decide later to drop or not.
+        /// </summary>
+        internal bool ShouldProcess(LoRaRequest loRaRequest, LoRaDevice loRaDevice)
+            => (loRaRequest.Payload is LoRaPayloadData && this.deduplicationStrategy.Create(loRaDevice) is DeduplicationStrategyDrop)
```

Comment:
## Problem
- Currently by default, deduplication strategy is None.
- Concentrator Deduplication in that case let's multiple requests pass through.
- The function does not set FrameCounterDown = 0
- We get a collision downstream

## Possible solutions
- Concentrator Deduplication drops if strategy is Drop **or** None, which is a bit weird.
- We introduce a different configuration for concentrator (vs "abusing" the one for the gateway/network server)

Reply:
The problem here is only happening in single gateway setup. If we do not drop the messages and the configuration for the device is having a gateway assigned, we do simply increment the framecounter on request, without validating if that request for that particular device was already handled. Hence, there we would need a strategy that takes multi concentrator messages into account.

Reply:
Not sending downstream was addressed by adding flag `skipDownstreamToAvoidCollisions`